admin/admin-service/src/main/java/io/mosip/admin/service/impl/ApplicantDetailServiceImpl.java

```diff
@@ -132,18 +133,22 @@ public ApplicantDetailsDto getApplicantDetails(String rid) throws Exception {
     public byte[] getRIDDigitalCard(String rid, boolean isAcknowledged) throws Exception {
         byte[] data=null;
         if(!isAcknowledged){
-            throw new RequestException(
+            throw new MasterDataServiceException(
                     ApplicantDetailErrorCode.DIGITAL_CARD_NOT_ACKNOWLEDGED.getErrorCode(),
                     ApplicantDetailErrorCode.DIGITAL_CARD_NOT_ACKNOWLEDGED.getErrorMessage());
         }
         DigitalCardStatusResponseDto digitalCardStatusResponseDto =getDigitialCardStatus(rid);
         if(!digitalCardStatusResponseDto.getStatusCode().equalsIgnoreCase(AVAILABLE)) {
             auditUtil.setAuditRequestDto(EventEnum.RID_DIGITAL_CARD_REQ_EXCEPTION);
-            throw new RequestException(
+            throw new MasterDataServiceException(
                     ApplicantDetailErrorCode.DIGITAL_CARD_RID_NOT_FOUND.getErrorCode(),
                     ApplicantDetailErrorCode.DIGITAL_CARD_RID_NOT_FOUND.getErrorMessage());
         }
         data = restClient.getForApi(digitalCardStatusResponseDto.getUrl(), byte[].class);
+        if(new String(data).contains("errors")) {
```

Comment:
You are trying to create a new instance of string every time no matter success or failure. This is not the correct way to check errors.

Reply:
taken care